zerver/lib/scim.py

```diff
@@ -270,6 +270,7 @@ def save(self) -> None:
                 password,
                 realm,
                 full_name_new_value,
+                tos_version=UserProfile.TOS_VERSION_BEFORE_FIRST_LOGIN,
```

Comment:
I do not have complete knowledge how authentication using `scim` and `ldap` works, but since we are just creating the user here, I have added this for now.

Reply:
I believe this is correct for SCIM, as well as LDAP, since in both cases, you haven't seen the registration page and accepted the terms of service or been presented with the email address visibility policy prompt.

There's a possible future where we add a setting for corporate organizations to not bother asking about email address visibility on signup because you're using your corporate email anyway that we should consider.

Reply:
@timabbott  Yeah that's right, the SCIM path is called when e.g. Okta server makes a request to Zulip to create the user (so no user interaction here at all) and the ldap path is called when the user is logging in for the first time, not having an account yet, and we just-in-time create it for them transparently.